server/src/test/java/org/elasticsearch/index/engine/NoopEngineTests.java

```diff
@@ -0,0 +1,105 @@
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.engine;
+
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.IndexReader;
+import org.elasticsearch.cluster.routing.AllocationId;
+import org.elasticsearch.cluster.routing.IndexShardRoutingTable;
+import org.elasticsearch.cluster.routing.ShardRouting;
+import org.elasticsearch.cluster.routing.ShardRoutingState;
+import org.elasticsearch.cluster.routing.TestShardRouting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.IndexSettings;
+import org.elasticsearch.index.mapper.ParsedDocument;
+import org.elasticsearch.index.seqno.ReplicationTracker;
+import org.elasticsearch.index.seqno.SequenceNumbers;
+import org.elasticsearch.test.IndexSettingsModule;
+
+import java.io.IOException;
+import java.util.Collections;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class NoopEngineTests extends EngineTestCase {
+    private static final IndexSettings INDEX_SETTINGS = IndexSettingsModule.newIndexSettings("index", Settings.EMPTY);
+
+    public void testNoopEngine() throws IOException {
+        engine.close();
+        final NoopEngine engine = new NoopEngine(noopConfig(INDEX_SETTINGS, store, primaryTranslogDir));
+        expectThrows(UnsupportedOperationException.class, () -> engine.index(null));
+        expectThrows(UnsupportedOperationException.class, () -> engine.delete(null));
+        expectThrows(UnsupportedOperationException.class, () -> engine.noOp(null));
+        expectThrows(UnsupportedOperationException.class, () -> engine.syncFlush(null, null));
+        expectThrows(UnsupportedOperationException.class, () -> engine.get(null, null));
+        expectThrows(UnsupportedOperationException.class, () -> engine.acquireSearcher(null, null));
+        expectThrows(UnsupportedOperationException.class, () -> engine.ensureTranslogSynced(null));
+        expectThrows(UnsupportedOperationException.class, engine::activateThrottling);
+        expectThrows(UnsupportedOperationException.class, engine::deactivateThrottling);
+        assertThat(engine.refreshNeeded(), equalTo(false));
+        assertThat(engine.shouldPeriodicallyFlush(), equalTo(false));
+        engine.close();
+    }
+
+    public void testTwoNoopEngines() throws IOException {
+        engine.close();
+        // It's so noop you can even open two engines for the same store without tripping anything
```

Comment:
Is this something that we are going to rely on?

Reply:
Probably not? I just thought it was a good test that we aren't acquiring locks or locking files accidentally or anything like that

Reply:
Okay, maybe the test name should be clearer, or at least a comment of its purpose?

Reply:
Sure I'll add a comment